server/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -889,4 +894,38 @@ public void testExistsWithFallback() {
         }
     }
 
+    public void testExclusionFilterDefaultsNotOmitted() {
+        final Setting.AffixSetting<String> setting = CLUSTER_ROUTING_EXCLUDE_GROUP_SETTING;
```

Comment:
I don't think we should use real settings in SettingTests. You can create a dummy setting here that happens to be similar to a real setting.

Reply:
You're totally right @rjernst, thank you. I've updated my pull request.

Reply:
I think that @rjernst suggestion was not about the value but about creating a dummy test setting (e.g., '`foo.bar`) that has similar properties to the routing settings. So rather than tying the test to a specific setting, we would want to be testing a generic setting with the properties of what you're trying to fix here.

Reply:
Got it! It would be far more better, yes. I'll working on it, thank you.

Reply:
I've updated my pull request and now both test cases are covered when setting a null value: defaults omitted or not.